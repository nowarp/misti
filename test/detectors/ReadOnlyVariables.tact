fun test1(): Int {
    let a: Int = 10; // Warning: read-only variable
    return a;
}

fun test2() {
    let a: Int = 10; // warning: read-only variable
    repeat (a) { /* do nothing */ }
    repeat (a) { /* ... */ }
}

fun test3() {
    let a: Int = 42; // warning: read-only variable
    if (a == 42) {}
}

fun test4() {
    let a: Int = 10; // warning: read-only variable
    if (a == 42) { /* do nothing */ }
}

fun test5() {
    let _a: Int = 10; // suppressed, no warning
    if (_a == 42) { /* do nothing */ }
}


contract Test {
    m: map<Int, Int>;
    get fun test(a: Int): Bool {
        let result: Int? = self.m.get(a); // OK: Cannot create a constant yet
        if (result == 1) {
            return true;
        }
        return false;
    }
}
