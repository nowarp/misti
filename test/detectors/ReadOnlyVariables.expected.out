[MEDIUM] ReadOnlyVariables: Read-only variable
test/detectors/ReadOnlyVariables.tact:7:9:
  6 | fun test2() {
> 7 |     let a: Int = 10; // warning: read-only variable
              ^
  8 |     repeat (a) { /* do nothing */ }
Help: Consider creating a constant instead
See: https://nowarp.io/tools/misti/docs/detectors/ReadOnlyVariables

[MEDIUM] ReadOnlyVariables: Read-only variable
test/detectors/ReadOnlyVariables.tact:13:9:
  12 | fun test3() {
> 13 |     let a: Int = 42; // warning: read-only variable
               ^
  14 |     if (a == 42) {}
Help: Consider creating a constant instead
See: https://nowarp.io/tools/misti/docs/detectors/ReadOnlyVariables

[MEDIUM] ReadOnlyVariables: Read-only variable
test/detectors/ReadOnlyVariables.tact:18:9:
  17 | fun test4() {
> 18 |     let a: Int = 10; // warning: read-only variable
               ^
  19 |     if (a == 42) { /* do nothing */ }
Help: Consider creating a constant instead
See: https://nowarp.io/tools/misti/docs/detectors/ReadOnlyVariables